packages/components/src/Sales/Direct/Form.tsx

```diff
@@ -124,15 +124,45 @@ const SalesDirectForm: VFC<Props> = ({
     return priceUnit.mul(feesPerTenThousand).div(10000)
   }, [price, priceUnit, feesPerTenThousand])
 
+  const amountFeesWithQuantity = useMemo(() => {
```

Comment:
Why not parsing the quantity to a BigNumber in a single memo and use the value of this memo to do the calculation at the bottom of this file?
```tsx
  const quantityBN = useMemo(() => {
    if (!quantity) return BigNumber.from(0)
    try {
      return BigNumber.from(quantity)
    } catch {
      console.error(`Cannot parse quantity as BigNumber with value ${quantity}`)
      return BigNumber.from(0)
    }
  }, [quantity])
```

Reply:
Updated